### PR TITLE
Set Hare tests in CI pipeline

### PR DESCRIPTION
**Pull request by beckmani**

### CODE REVIEW

File: tests/hare/test_hare.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def query_hare_output_set(indx, namespace, client_po_name):
 
 
 def test_hare_sanity(load_config, setup_clients, save_log_on_exit):
-    delay = int(testconfig['client']['args']['hare-round-duration-sec']) * 7
+    delay = int(testconfig['client']['args']['hare-round-duration-sec']) * 8
```
Comment (antonlerner) [inline tests/hare/test_hare.py]: why 8? whats 8? name the magic num

File: tests/test_bs.py [hunk 2]
```diff
@@ -351,7 +351,7 @@ def test_gossip(load_config, setup_clients):
     (out, err) = p.communicate()
     assert '{"value":"ok"}' in out.decode("utf-8")
 
-    gossip_propagation_sleep = len(setup_clients.pods) * 2
+    gossip_propagation_sleep = len(setup_clients.pods)
```
Comment (antonlerner) [inline tests/test_bs.py:355]: comment to explain

File: Makefile [hunk 3]
```diff
@@ -125,6 +126,8 @@ dockerpush: dockerbuild-go
 
 	docker push spacemeshos/$(DOCKER_IMAGE_NAME):$(BRANCH)
 	docker push spacemeshos/$(DOCKER_IMAGE_NAME):$(SHA)
+	docker tag $(DOCKER_IMAGE_NAME):$(BRANCH) spacemeshos/$(TESTS_DOCKER_IMAGE_NAME):$(BRANCH)
```
Comment (antonlerner) [inline Makefile]: rename DOCKER_IMAGE_NAME to DOCKER_REPO or something

File: tests/test_bs.py (hunk 2 above)
Comment (beckmani) [reply tests/test_bs.py:355]: Initially it was 2 sec delay for each spacemesh node. 
This is only assessment

File: tests/hare/test_hare.py (hunk 1 above)
Comment (beckmani) [reply tests/hare/test_hare.py]: This is only assessment to the delay before checking that all nodes report in ES.
@gavraz any comments on that?

Comment (gavraz) [reply tests/hare/test_hare.py]: Yes. It should be 5. You can name it NUM_OF_EXPECTED_ROUNDS

File: tests/test_bs.py (hunk 2 above)
Comment (antonlerner) [reply tests/test_bs.py:355]: add comment in code to explain this...

File: Makefile (hunk 3 above)
Comment (beckmani) [reply Makefile]: Done in: https://github.com/spacemeshos/go-spacemesh/pull/773

File: Makefile [hunk 4]
```diff
@@ -140,4 +143,5 @@ ifndef ES_PASSWD
 	$(error ES_PASSWD is not set)
 endif
 	docker run -e ES_PASSWD="$(ES_PASSWD)" -e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json -it go-spacemesh-python pytest -s test_bs.py --tc-file=config.yaml --tc-format=yaml
+	docker run -e ES_PASSWD="$(ES_PASSWD)" -e GOOGLE_APPLICATION_CREDENTIALS=./spacemesh.json -it go-spacemesh-python pytest -s hare/test_hare.py --tc-file=hare/config.yaml --tc-format=yaml
```
Comment (gavraz) [inline Makefile:148]: You should probably make a script to run this for all tests. All tests should have the same structure.

**Review by gavraz — APPROVED** [review]
Please make sure the tests are working after the changes.

File: tests/test_bs.py (hunk 2 above)
Comment (beckmani) [reply tests/test_bs.py:355]: Done

File: Makefile [hunk 5]
```diff
@@ -121,13 +124,12 @@ dockerbuild-go:
...
 dockerpush: dockerbuild-go
 	echo "$(DOCKER_PASSWORD)" | docker login -u "$(DOCKER_USERNAME)" --password-stdin
-	docker tag $(DOCKER_IMAGE_NAME):$(BRANCH) spacemeshos/$(DOCKER_IMAGE_NAME):$(BRANCH)
-	docker tag $(DOCKER_IMAGE_NAME):$(BRANCH) spacemeshos/$(DOCKER_IMAGE_NAME):$(SHA)
-
-	docker push spacemeshos/$(DOCKER_IMAGE_NAME):$(BRANCH)
-	docker push spacemeshos/$(DOCKER_IMAGE_NAME):$(SHA)
-	docker tag $(DOCKER_IMAGE_NAME):$(BRANCH) spacemeshos/$(TESTS_DOCKER_IMAGE_NAME):$(BRANCH)
-	docker push spacemeshos/$(TESTS_DOCKER_IMAGE_NAME):$(BRANCH)
+	docker tag $(DOCKER_IMAGE_REPO):$(BRANCH) spacemeshos/$(DOCKER_IMAGE_REPO):$(BRANCH)
```
Comment (gavraz) [inline Makefile]: consider introducing a function to tag and push

**Review by gavraz — APPROVED** [review]

**Pull request merged**